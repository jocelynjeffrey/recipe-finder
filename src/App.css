.App {
  text-align: center;
  display: flex;
  flex-wrap: wrap;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
  width: 100%;
}

.App-intro {
  font-size: large;
}

.App-body {
  width: 100%;
  border:1px solid red;
  display: flex;
  flex-wrap: wrap;
}

.search-component {
  border: 2px solid purple;
  height: 50px;
  width: 100%;
}

.results-component {
  border: 1px solid blue;
  width: 50%;
  box-sizing: border-box;
}

.result {
  width: 100%;
  background-color: #fff;
  border: none;
}

.result:hover {
  background-color: lightblue;
}

.details-component {
  border: 1px solid orange;
  width: 50%;
  box-sizing: border-box;
}