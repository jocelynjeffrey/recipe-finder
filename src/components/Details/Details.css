
.details-component {
  display: flex;
  flex-direction: column;
  padding: 0 25px 25px;
  width: 60%;
  box-sizing: border-box;
}

.details-component h3 {
  padding-top: 20px;
  margin: 10px 0;
}

.details-component img {
  border: 1px solid #C2C2C4;
  width: 100%;
  max-width: 400px;
  margin: auto;
}

.details-component .btn {
  cursor: pointer;
  color: #fff;
  text-decoration: none;
  width: 60%;
  display: block;
  margin: 30px auto;
  background-color: #EF4581;
  font-weight: 700;
  border: none;
  font-size: 18px;
  padding: 10px 5px;
}

.details-component .btn:hover {
  background-color: #EF4581;
}

.details-component ul {
  list-style: none;
  margin: 0;
  padding: 0 20px;
}

.details-component li {
  padding: 5px 0;
}
