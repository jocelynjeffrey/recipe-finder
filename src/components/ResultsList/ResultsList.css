.results-component {
  width: 40%;
  display: block;
  box-sizing: border-box;
  background-color: #E6E7E8;
}

.results-component .btn {
  width: 100%;
  border: none;
  background-color: #E6E7E8;
  box-sizing: border-box;
  font-weight: 700;
  font-size: 14px;
  min-height: 50px;
  height: inherit;
}

.results-component .btn:hover {
  background-color: #F9A11F;
}

.results-component .btn:focus {
  background-color: #80cbc4;
  outline: 0;
}

.no-results {
  text-align: center;
  width: 100%;
}
