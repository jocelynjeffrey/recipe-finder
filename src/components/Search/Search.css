.search-component {
  background-color: #E6E7E8;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  padding: 20px;
  width: 100%;
  font-size: large;
}

.search-component input {
  width: 100%;
  padding: 0 10px;
  border: none;
  height: 40px;
  font-size: 16px;
  margin: 10px 0;
}

.search-component label {
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  padding-right: 10px;
}

.search-component button {
  width: 70%;
  color: #fff;
  margin-top: 15px;
}

.search-component button:disabled {
  color: #E6E7E8;
  background-color: #C2C2C4;
}

.toggle-tabs {
  display: flex;
  width: 100%;
  justify-content: center;
}

.toggle-tabs a {
  cursor: pointer;
  font-weight: 700;
  width: 30%;
  max-width: 300px;
  padding: 15px 10px;
  color: #E6E7E8;
  background-color: #C2C2C4;
  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
}

.toggle-tabs a:first-of-type {
  margin-right: -15px;
}

.toggle-tabs a:hover {
  background-color: #80cbc4;
  color: #fff;
}

.toggle-tabs a.active {
  color: #fff;
  font-weight: 700;
  background-color:#01848F;
  cursor: auto;
  z-index: 1;
}

.search-component .form {
  width: 80%;
  padding: 20px 25px;
  max-width: 615px;
  border: 1px solid #C2C2C4;
  border-radius: 5px;
}

.search-component .form div {
  justify-content: flex-start;
}

.search-tab {
  display: flex;
  flex-wrap: wrap;
}

.calorie-tabs {
  display: none;
}

.calorie-tabs label {
  width: 100%;
  text-align: left;
}

.calorie-tabs.active {
  display: flex;
  flex-wrap: wrap;
}

.calorie-input {
  display: none;
}

.calorie-input.active:first-of-type {
  margin-right: 10px;
}

.calorie-input.active {
  display: inline-block;
  width: 60px;
}