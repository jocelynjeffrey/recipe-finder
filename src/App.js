import React from 'react';
import './App.css';
import { connect } from 'react-redux';
import { getRecipeByName, onTextInputChange, showRecipeDetails } from './redux/actions/recipe.actions'

const SearchComponent = props => (
  <div className="search-component">
    <input
      value={props.textInput}
      onChange={e => props.onTextInputChange(e.target.value)}
    />
    <button onClick={ () => {props.getRecipeByName(props.textInput)} }>Search</button>
  </div>
);

const Details = props => {
  if (props.selectedRecipe) {
    return <div className="details-component">
      <h2>{props.selectedRecipe.label}</h2>
      <img src={props.selectedRecipe.image} />
      <ul>
        {props.selectedRecipe.ingredients.map(ingredient =>
            <li>{ingredient.text}</li>
          )}
      </ul>
    </div>
  }
  return <p>select a recipe</p>

}

const ResultsList = props => (
  <div className="results-component">
    {props.recipeList.map(recipe =>
      <button
      className='result'
      key={recipe.uri}
      onClick={ () => { props.showRecipeDetails(recipe) } }>
        <h3>{recipe.label} </h3>
      </button>
    )}
  </div>
);

const App = props => (
  <div className="App">
    <div className="App-header">
      <h2>Recipe Finder</h2>
    </div>
    <div className="App-body">
      <SearchComponent {...props} />
      <ResultsList {...props} />
      <Details {...props} />
    </div>
  </div>
);

const mapStateToProps = (state) => {
  console.log('state', state)
  return {
    textInput: state.inputChange.textInput,
    recipeList: state.recipe.list,
    selectedRecipe: state.recipe.selectedRecipe,
  }
}

const actions = {
  onTextInputChange,
  getRecipeByName,
  showRecipeDetails,
}

export default connect(
  mapStateToProps,
  actions
)(App);