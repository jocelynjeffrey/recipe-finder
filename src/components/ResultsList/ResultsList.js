import React from 'react';
import '../../App.css';
import FavouritesList from '../FavouritesList';

const ResultsList = props => {
  { if (props.recipeList.length === 0) return null; }
  return <div className="results-component">
    <h2>Select a Recipe</h2>
    {props.recipeList.map(recipe =>
      <button
      className='btn'
      key={recipe.uri}
      onClick={ () => { props.showRecipeDetails(recipe) } }>
        {recipe.label}
      </button>
    )}
    <FavouritesList {...props} />
  </div>
};

export default ResultsList;
