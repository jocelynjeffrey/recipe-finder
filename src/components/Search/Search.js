import React from 'react';
import '../../App.css';

const Search = props => (
  <div className="search-component">
    <input
      value={props.textInput}
      onChange={e => props.onTextInputChange(e.target.value)}
    />
    <button onClick={ () => {props.getRecipeByName(props.textInput)} }>Search</button>
  </div>
);

export default Search;