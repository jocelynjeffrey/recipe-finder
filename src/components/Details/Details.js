import React from 'react';
import LikeButton from '../LikeButton';

import '../../App.css';
import './Details.css';

const Details = props => {
  if (props.selectedRecipe) {
    return <div className="details-component">
    <LikeButton {...props} />
      <h2>{props.selectedRecipe.label}</h2>
      <img src={props.selectedRecipe.image} />
      <h3>Ingredients</h3>
      <ul>
        {props.selectedRecipe.ingredients.map((ingredient, i) =>
            <li key={i}>{ingredient.text}</li>
          )}
      </ul>
      <h3>Calories</h3>
      <span>{Math.round(props.selectedRecipe.calories)}</span>
      <h3>Source</h3>
      <span>{props.selectedRecipe.source}</span>
      <a className="btn" target="_blank" href={props.selectedRecipe.url}>Go to Instructions</a>
    </div>
  }
  return null
}

export default Details;
