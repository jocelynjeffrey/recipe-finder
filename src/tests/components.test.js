import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { App } from '../App';
import Details from '../components/Details';
import ResultsList from '../components/ResultsList';
import Search from '../components/Search';
import { shallow } from 'enzyme';

describe('component tests', function() {

  const mockData = {
    recipeList: [ {label: 'coffee', uri: 1}, {label: 'tea', uri: 2} ],
    isError: false,
    selectedRecipe: {label: 'coffee', calories: 3},
    textInput: 'pineapple',
  }

  describe('<App />', function() {
    it('renders correctly', function() {
      const wrapper = shallow(<App {...mockData} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('<Details />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Details />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('<ResultsList />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<ResultsList {...mockData} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('<Search />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Search />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });


});
