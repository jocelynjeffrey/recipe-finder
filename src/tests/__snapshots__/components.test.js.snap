// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component tests <App /> renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App"
>
    <div
        className="App-header"
    >
        <h1>
            Recipe Finder
        </h1>
    </div>
    <Search
        isError={false}
        recipeList={
            Array [
                Object {
                  "label": "coffee",
                  "uri": 1,
                },
                Object {
                  "label": "tea",
                  "uri": 2,
                },
              ]
        }
        selectedRecipe={
            Object {
                "calories": 3,
                "label": "coffee",
              }
        }
        textInput="pineapple"
    />
    <div
        className="App-body"
    >
        <ResultsList
            isError={false}
            recipeList={
                Array [
                    Object {
                      "label": "coffee",
                      "uri": 1,
                    },
                    Object {
                      "label": "tea",
                      "uri": 2,
                    },
                  ]
            }
            selectedRecipe={
                Object {
                    "calories": 3,
                    "label": "coffee",
                  }
            }
            textInput="pineapple"
        />
        <Details
            isError={false}
            recipeList={
                Array [
                    Object {
                      "label": "coffee",
                      "uri": 1,
                    },
                    Object {
                      "label": "tea",
                      "uri": 2,
                    },
                  ]
            }
            selectedRecipe={
                Object {
                    "calories": 3,
                    "label": "coffee",
                  }
            }
            textInput="pineapple"
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="App"
>
      <div
            className="App-header"
      >
            <h1>
                  Recipe Finder
            </h1>
      </div>
      <Search
            isError={false}
            recipeList={
                  Array [
                        Object {
                          "label": "coffee",
                          "uri": 1,
                        },
                        Object {
                          "label": "tea",
                          "uri": 2,
                        },
                      ]
            }
            selectedRecipe={
                  Object {
                        "calories": 3,
                        "label": "coffee",
                      }
            }
            textInput="pineapple"
      />
      <div
            className="App-body"
      >
            <ResultsList
                  isError={false}
                  recipeList={
                        Array [
                              Object {
                                "label": "coffee",
                                "uri": 1,
                              },
                              Object {
                                "label": "tea",
                                "uri": 2,
                              },
                            ]
                  }
                  selectedRecipe={
                        Object {
                              "calories": 3,
                              "label": "coffee",
                            }
                  }
                  textInput="pineapple"
            />
            <Details
                  isError={false}
                  recipeList={
                        Array [
                              Object {
                                "label": "coffee",
                                "uri": 1,
                              },
                              Object {
                                "label": "tea",
                                "uri": 2,
                              },
                            ]
                  }
                  selectedRecipe={
                        Object {
                              "calories": 3,
                              "label": "coffee",
                            }
                  }
                  textInput="pineapple"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        isError={false}
        recipeList={
                Array [
                        Object {
                          "label": "coffee",
                          "uri": 1,
                        },
                        Object {
                          "label": "tea",
                          "uri": 2,
                        },
                      ]
        }
        selectedRecipe={
                Object {
                        "calories": 3,
                        "label": "coffee",
                      }
        }
        textInput="pineapple"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isError": false,
          "recipeList": Array [
            Object {
              "label": "coffee",
              "uri": 1,
            },
            Object {
              "label": "tea",
              "uri": 2,
            },
          ],
          "selectedRecipe": Object {
            "calories": 3,
            "label": "coffee",
          },
          "textInput": "pineapple",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <h1>
                              Recipe Finder
                    </h1>
          </div>
          <Search
                    isError={false}
                    recipeList={
                              Array [
                                        Object {
                                          "label": "coffee",
                                          "uri": 1,
                                        },
                                        Object {
                                          "label": "tea",
                                          "uri": 2,
                                        },
                                      ]
                    }
                    selectedRecipe={
                              Object {
                                        "calories": 3,
                                        "label": "coffee",
                                      }
                    }
                    textInput="pineapple"
          />
          <div
                    className="App-body"
          >
                    <ResultsList
                              isError={false}
                              recipeList={
                                        Array [
                                                  Object {
                                                    "label": "coffee",
                                                    "uri": 1,
                                                  },
                                                  Object {
                                                    "label": "tea",
                                                    "uri": 2,
                                                  },
                                                ]
                              }
                              selectedRecipe={
                                        Object {
                                                  "calories": 3,
                                                  "label": "coffee",
                                                }
                              }
                              textInput="pineapple"
                    />
                    <Details
                              isError={false}
                              recipeList={
                                        Array [
                                                  Object {
                                                    "label": "coffee",
                                                    "uri": 1,
                                                  },
                                                  Object {
                                                    "label": "tea",
                                                    "uri": 2,
                                                  },
                                                ]
                              }
                              selectedRecipe={
                                        Object {
                                                  "calories": 3,
                                                  "label": "coffee",
                                                }
                              }
                              textInput="pineapple"
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <h1>
                              Recipe Finder
                    </h1>
          </div>
          <Search
                    isError={false}
                    recipeList={
                              Array [
                                        Object {
                                          "label": "coffee",
                                          "uri": 1,
                                        },
                                        Object {
                                          "label": "tea",
                                          "uri": 2,
                                        },
                                      ]
                    }
                    selectedRecipe={
                              Object {
                                        "calories": 3,
                                        "label": "coffee",
                                      }
                    }
                    textInput="pineapple"
          />
          <div
                    className="App-body"
          >
                    <ResultsList
                              isError={false}
                              recipeList={
                                        Array [
                                                  Object {
                                                    "label": "coffee",
                                                    "uri": 1,
                                                  },
                                                  Object {
                                                    "label": "tea",
                                                    "uri": 2,
                                                  },
                                                ]
                              }
                              selectedRecipe={
                                        Object {
                                                  "calories": 3,
                                                  "label": "coffee",
                                                }
                              }
                              textInput="pineapple"
                    />
                    <Details
                              isError={false}
                              recipeList={
                                        Array [
                                                  Object {
                                                    "label": "coffee",
                                                    "uri": 1,
                                                  },
                                                  Object {
                                                    "label": "tea",
                                                    "uri": 2,
                                                  },
                                                ]
                              }
                              selectedRecipe={
                                        Object {
                                                  "calories": 3,
                                                  "label": "coffee",
                                                }
                              }
                              textInput="pineapple"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    isError={false}
    recipeList={
        Array [
            Object {
              "label": "coffee",
              "uri": 1,
            },
            Object {
              "label": "tea",
              "uri": 2,
            },
          ]
    }
    selectedRecipe={
        Object {
            "calories": 3,
            "label": "coffee",
          }
    }
    textInput="pineapple"
/>,
}
`;

exports[`component tests <Details /> renders correctly 1`] = `<span />`;

exports[`component tests <ResultsList /> renders correctly 1`] = `
<div
  className="results-component"
>
  <button
    className="result"
    onClick={[Function]}
  >
    coffee
  </button>
  <button
    className="result"
    onClick={[Function]}
  >
    tea
  </button>
</div>
`;

exports[`component tests <Search /> renders correctly 1`] = `
<div
  className="search-component"
>
  <input
    onChange={[Function]}
    value={undefined}
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
